### PR TITLE
Quick Bar: Use command category labels instead of icons

### PR DESCRIPTION
**Pull request by donkawechico**
## Proposed change

Fix several issues related to translations and capitalization by implementing a feature that was likely going to be added anyway:

Removing command icons in favor of stylized command "category" labels. Similar to chrome devtools command palette.

## Considerations

As of this writing, this PR would also remove the word "Reload" from the configuration page's "YAML Configuration Reload" section. This might be okay, though, since each item is already within a section labeled "Reload":

We could re-add the word "Reload" but it would mean either maintaining two sets of near-duplicate strings, or adding new explicit "Reload *" strings for every other supported reload service.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
Please use material chips for the labels

**Comment by spacegaier** [comment]
@donkawechico Will you have time / motivation to tackle that soon-ish? Otherwise I might jump in and incorporate Brams request, because I really would like to see this merged and then I would want to add additional commands.

**Comment by donkawechico** [comment]
> @donkawechico Will you have time / motivation to tackle that soon-ish? Otherwise I might jump in and incorporate Brams request, because I really would like to see this merged and then I would want to add additional commands.

I was actually planning on starting to contribute again this week. Lemme see if I can get to this this week.

**Comment by donkawechico** [comment]
Looking at this today/tonight.

**Comment by donkawechico** [comment]
@bramkragten This could use a review when you have some time.

It's been awhile since I worked on this so I may have missed some things. But the chips look right in command bar and in the devices tab (where the automation and trigger cards also use `ha-chip-set`)

File: hassio/src/dialogs/network/dialog-hassio-network.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ import {
 } from "lit-element";
 import { cache } from "lit-html/directives/cache";
 import { fireEvent } from "../../../../src/common/dom/fire_event";
-import "../../../../src/components/ha-chips";
+import "../../../../src/components/ha-chip-set";
```
Comment (bramkragten) [inline hassio/src/dialogs/network/dialog-hassio-network.ts:21]:
```suggestion
```

File: src/components/ha-chip-set.ts [hunk 2]
```diff
@@ -60,16 +51,16 @@ export class HaChips extends LitElement {
   static get styles(): CSSResult {
     return css`
       ${unsafeCSS(chipStyles)}
-      .mdc-chip {
-        background-color: rgba(var(--rgb-primary-text-color), 0.15);
-        color: var(--primary-text-color);
+      ha-chip {
+        --ha-chip-background-color: rgba(var(--rgb-primary-text-color), 0.15);
+        --ha-chip-text-color: var(--primary-text-color);
```
Comment (bramkragten) [inline src/components/ha-chip-set.ts:56]: This should be the default of `ha-chip`, and not be overruled by the set.
```suggestion
```

File: src/components/ha-chip.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    `;
+  }
+
+  static get styles(): CSSResult {
+    return css`
+      ${unsafeCSS(chipStyles)}
+      .mdc-chip {
+        margin: 4px;
+        background-color: var(--ha-chip-background-color);
+        color: var(--ha-chip-text-color);
```
Comment (bramkragten) [inline src/components/ha-chip.ts:45]:
```suggestion
        background-color: var(--ha-chip-background-color, rgba(var(--rgb-primary-text-color), 0.15));
        color: var(--ha-chip-text-color, var(--primary-text-color));
```

File: src/dialogs/quick-bar/ha-quick-bar.ts [hunk 4]
```diff
@@ -516,6 +585,27 @@ export class QuickBar extends LitElement {
...
+          --ha-chip-background-color: pink;
+        }
+
+        .command-category.navigation {
+          --ha-chip-background-color: lightblue;
+        }
+
+        .command-category.server_control {
+          --ha-chip-background-color: orange;
+          --ha-chip-text-color: white;
```
Comment (bramkragten) [inline src/dialogs/quick-bar/ha-quick-bar.ts:606]:
```suggestion
          --ha-chip-background-color: var(--warning-color);
```

File: src/dialogs/quick-bar/ha-quick-bar.ts [hunk 5]
```diff
@@ -516,6 +585,27 @@ export class QuickBar extends LitElement {
...
+          border-radius: 6px;
+          color: black;
+        }
+
+        .command-category.reload {
+          --ha-chip-background-color: pink;
+        }
+
+        .command-category.navigation {
+          --ha-chip-background-color: lightblue;
```
Comment (bramkragten) [inline src/dialogs/quick-bar/ha-quick-bar.ts:601]:
```suggestion
          --ha-chip-background-color: var(--light-primary-color);
```

File: src/dialogs/quick-bar/ha-quick-bar.ts [hunk 6]
```diff
@@ -516,6 +585,27 @@ export class QuickBar extends LitElement {
...
+        span.command-category {
+          font-weight: bold;
+          padding: 3px;
+          display: inline-flex;
+          border-radius: 6px;
+          color: black;
+        }
+
+        .command-category.reload {
+          --ha-chip-background-color: pink;
```
Comment (bramkragten) [inline src/dialogs/quick-bar/ha-quick-bar.ts:597]: We should probably make this one themeable too...
```suggestion
          --ha-chip-background-color: #cddc39;
```

File: src/resources/styles.ts [hunk 7]
```diff
@@ -92,6 +92,7 @@ export const derivedStyles = {
   "mdc-button-disabled-ink-color": "var(--disabled-text-color)",
   "mdc-button-outline-color": "var(--divider-color)",
   "mdc-dialog-scroll-divider-color": "var(--divider-color)",
+  "chip-background-color": "rgba(var(--rgb-primary-text-color), 0.15)",
```
Comment (bramkragten) [inline src/resources/styles.ts:95]:
```suggestion
```

**Comment by bramkragten** [comment]
I noticed that `> ` in the input yields zero results, when I remove the space `>` gives me all options.

File: src/dialogs/quick-bar/ha-quick-bar.ts [hunk 8]
```diff
@@ -315,94 +362,112 @@ export class QuickBar extends LitElement {
...
+        `ui.dialogs.quick-bar.commands.types.reload`
+      );
+      const primaryText =
         this.hass.localize(`ui.dialogs.quick-bar.commands.reload.${domain}`) ||
         this.hass.localize(
           "ui.dialogs.quick-bar.commands.reload.reload",
           "domain",
           domainToName(this.hass.localize, domain)
-        ),
-      icon: domainIcon(domain),
```
Comment (bramkragten) [inline src/dialogs/quick-bar/ha-quick-bar.ts:346]: Should we keep the icon and add it to the chip? 
image

Comment (donkawechico) [reply src/dialogs/quick-bar/ha-quick-bar.ts:346]: Ooooo... I like that. Sure

**Comment by donkawechico** [comment]
> I noticed that `> ` in the input yields zero results, when I remove the space `>` gives me all options.

I remember having a discussion about this months ago with someone... maybe you?... but I forget what the outcome was. I'll take a look

File: src/dialogs/quick-bar/ha-quick-bar.ts [hunk 9]
```diff
@@ -516,6 +585,27 @@ export class QuickBar extends LitElement {
...
+        span.command-category {
+          font-weight: bold;
+          padding: 3px;
+          display: inline-flex;
+          border-radius: 6px;
+          color: black;
+        }
+
+        .command-category.reload {
+          --ha-chip-background-color: #cddc39;
```
Comment (donkawechico) [inline src/dialogs/quick-bar/ha-quick-bar.ts:597]: @bramkragten I committed your change here (changing `pink` to `#cddc39`) without checking what `#cddc39` was. Did you mean to choose a yellow here to replace the pink?

image

**Comment by homeassistant** [comment]
Hi @GitHub-Action, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#7692).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: src/dialogs/quick-bar/ha-quick-bar.ts (hunk 9 above)
Comment (bramkragten) [reply src/dialogs/quick-bar/ha-quick-bar.ts:597]: That's green right? 😄 Yeah, I thought it would match reload well?

**Comment by bramkragten** [comment]
> > I noticed that `> ` in the input yields zero results, when I remove the space `>` gives me all options.
> 
> I remember having a discussion about this months ago with someone... maybe you?... but I forget what the outcome was. I'll take a look

Oh, it is an existing issue... Should be a different PR btw 🙂

File: src/dialogs/quick-bar/ha-quick-bar.ts (hunk 9 above)
Comment (donkawechico) [reply src/dialogs/quick-bar/ha-quick-bar.ts:597]: It's much closer to yellow ;)

green == ( 0, 255, 0 )
yellow == ( 255, 255, 0 )
#cddc39 == ( 205, 255, 57 )

**Comment by homeassistant** [comment]
Hi @GitHub-Action, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#7692).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: src/dialogs/quick-bar/ha-quick-bar.ts (hunk 9 above)
Comment (donkawechico) [reply src/dialogs/quick-bar/ha-quick-bar.ts:615]: But yeah, I'm fine with it. For whatever reason, a pinkish color felt right to me so I guess it's just subjective. 🤷

File: src/dialogs/quick-bar/ha-quick-bar.ts (hunk 8 above)
Comment (donkawechico) [reply src/dialogs/quick-bar/ha-quick-bar.ts:346]: So, I tried keeping the original icons, but those are all based on the "noun" of the command (automation, script, etc). 

This made sense when we didn't have a category concept, but looks very confusing when you add them to chips of the same color and category.

So I instead used icons representing the broad category (like a circle symbol for all "Reload ____" commands). That looks much better, imho.

**Comment by homeassistant** [comment]
Hi @GitHub-Action, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#7692).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: src/dialogs/quick-bar/ha-quick-bar.ts [hunk 10]
```diff
@@ -516,6 +598,29 @@ export class QuickBar extends LitElement {
...
+          --ha-chip-text-color: black;
+        }
+
+        .command-category.navigation {
+          --ha-chip-background-color: var(--light-primary-color);
+          --ha-chip-text-color: black;
```
Comment (donkawechico) [inline src/dialogs/quick-bar/ha-quick-bar.ts:616]: @bramkragten Heads up that I had to force text color to `black` since dark mode changes primary text color to an unreadable shade of gray.

Is there a hard-coded dark shade more appropriate to HA than `black` that I should use?

File: src/panels/config/automation/action/types/ha-automation-action-delay.ts [hunk 11]
```diff
@@ -14,45 +21,57 @@ export class HaDelayAction extends LitElement implements ActionElement {
...
+    }
+    // Check for templates in action. If found, revert to YAML mode.
+    if (this.action && hasTemplate(this.action)) {
+      fireEvent(
+        this,
+        "ui-mode-not-available",
+        Error(this.hass.localize("ui.errors.config.no_template_editor_support"))
+      );
+      return;
+    }
```
Comment (donkawechico) [inline src/panels/config/automation/action/types/ha-automation-action-delay.ts:42]: Uhhhhhh why is this file changed.... crap did I mess up a rebase?

Comment (donkawechico) [reply src/panels/config/automation/action/types/ha-automation-action-delay.ts:42]: Yeah... I think I rebased with master instead of dev. Damned work habits. 🙄

File: src/dialogs/quick-bar/ha-quick-bar.ts (hunk 10 above)
Comment (bramkragten) [reply src/dialogs/quick-bar/ha-quick-bar.ts:616]: Just hard code the light theme primary text color

File: src/components/ha-chip.ts [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+      <div class="mdc-chip" .index=${this.index}>
+        ${this.hasIcon
+          ? html`<div class="mdc-chip__icon mdc-chip__icon--leading">
+              <slot name="icon"></slot>
+            </div>`
+          : null}
+        <div class="mdc-chip__ripple" .ripple="${ripple()}"></div>
+        <span role="gridcell">
+          <span role="button" tabindex="0" class="mdc-chip__primary-action">
+            <span class="mdc-chip__text"><slot name="label"></slot></span>
```
Comment (bramkragten) [inline src/components/ha-chip.ts:40]:
```suggestion
            <span class="mdc-chip__text"><slot></slot></span>
```

File: src/components/ha-chip-set.ts [hunk 13]
```diff
@@ -33,18 +33,9 @@ export class HaChips extends LitElement {
...
-                    role="button"
-                    tabindex="0"
-                    class="mdc-chip__primary-action"
-                  >
-                    <span class="mdc-chip__text">${item}</span>
-                  </span>
-                </span>
-              </div>
+              <ha-chip .index=${idx} @click=${this._handleClick}>
+                <span slot="label">${item}</span>
```
Comment (bramkragten) [inline src/components/ha-chip-set.ts:37]:
```suggestion
                <span slot="label">${item}</span>
```
```suggestion
                ${item}
```

File: src/components/ha-chip.ts [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+        </span>
+      </div>
+    `;
+  }
+
+  static get styles(): CSSResult {
+    return css`
+      ${unsafeCSS(chipStyles)}
+      .mdc-chip {
+        margin: 4px;
```
Comment (bramkragten) [inline src/components/ha-chip.ts:51]: Why do we have this margin now?

File: src/dialogs/quick-bar/ha-quick-bar.ts [hunk 15]
```diff
@@ -213,10 +232,50 @@ export class QuickBar extends LitElement {
...
+            .hasIcon=${true}
+            class="command-category ${item.categoryKey}"
+          >
+            ${item.iconPath
+              ? html`<ha-svg-icon
+                  .path=${item.iconPath}
+                  slot="icon"
+                ></ha-svg-icon>`
+              : ""}
+            <span slot="label">${item.categoryText}</span></ha-chip
```
Comment (bramkragten) [inline src/dialogs/quick-bar/ha-quick-bar.ts:274]:
```suggestion
            ${item.categoryText}</ha-chip
```

File: src/dialogs/quick-bar/ha-quick-bar.ts [hunk 16]
```diff
@@ -213,10 +232,50 @@ export class QuickBar extends LitElement {
...
+    return html`
+      <mwc-list-item
+        .twoline=${Boolean(item.altText)}
+        .item=${item}
+        index=${ifDefined(index)}
+      >
+        <span>
+          <ha-chip
+            .label="${item.categoryText}"
+            .hasIcon=${true}
```
Comment (bramkragten) [inline src/dialogs/quick-bar/ha-quick-bar.ts:265]:
```suggestion
            hasIcon
```

File: src/components/ha-chip.ts [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+  interface HASSDomEvents {
+    "chip-clicked": { index: string };
+  }
+}
+
+@customElement("ha-chip")
+export class HaChip extends LitElement {
+  @property() public index = 0;
+
+  @property() public hasIcon = false;
```
Comment (bramkragten) [inline src/components/ha-chip.ts:27]:
```suggestion
  @property({type: Boolean}) public hasIcon = false;
```

File: src/dialogs/quick-bar/ha-quick-bar.ts (hunk 16 above)
Comment (bramkragten) [inline src/dialogs/quick-bar/ha-quick-bar.ts:265]:
```suggestion
            .hasIcon=${item.iconPath}
```

File: src/components/ha-chip.ts (hunk 14 above)
Comment (donkawechico) [reply src/components/ha-chip.ts:51]: 🤷

Comment (donkawechico) [reply src/components/ha-chip.ts:51]: Ah I remember. I moved that margin styling from device automation card into here. The automation card needs the margins or else the chips get all squished.

But I'm sure I can fix that within `ha-device-automation-card`

Comment (donkawechico) [reply src/components/ha-chip.ts:51]: Done

**Review by bramkragten — APPROVED** [review]

File: src/components/ha-chip.ts (hunk 14 above)
Comment (bramkragten) [reply src/components/ha-chip.ts:51]: Done but not pushed or...? 😝

Comment (donkawechico) [reply src/components/ha-chip.ts:51]: Sigh... not sure how it didn't get in my recent push. Fixed here: https://github.com/home-assistant/frontend/pull/8718
